### ISSUE TITLE
How to use "phy_attack" dataset

### ISSUE THREAD

**Issue by pianoforte23** [idrl-lab/Full-coverage-camouflage-adversarial-attack#1]
Hello,
I found that the default dataset used in your code is "phy_attack" which is the same one used in DAS.
But I don't know how to set other directories such as "train_new", "train_label_new", etc.
Could you give me more detailed guidelines for using the dataset?

I'd really appreciate it if you reply to my question.
Thank you.

**Comment by zytxxn** [comment]
> Hello, I found that the default dataset used in your code is "phy_attack" which is the same one used in DAS. But I don't know how to set other directories such as "train_new", "train_label_new", etc. Could you give me more detailed guidelines for using the dataset?
> 
> I'd really appreciate it if you reply to my question. Thank you.

Hello,have you solved it yet?I can't find the dataset.

**Comment by pianoforte23** [comment]
> 

No, I didn't solve it yet :(

**Comment by winterwindwang** [comment]
> > 
> 
> No, I didn't solve it yet :(

Hey, we now provide the "train_label_new" file in [https://pan.baidu.com/s/18ZcQQgsr1DFbyUK4YVQ8WQ](https://pan.baidu.com/s/18ZcQQgsr1DFbyUK4YVQ8WQ) with the fetech code `idrl`

**Comment by winterwindwang** [comment]
> Hello, I found that the default dataset used in your code is "phy_attack" which is the same one used in DAS. But I don't know how to set other directories such as "train_new", "train_label_new", etc. Could you give me more detailed guidelines for using the dataset?
> 
> I'd really appreciate it if you reply to my question. Thank you.

You can obtain the "train_new" by fetech the image from the "DAS' dataset. And we provide the "train_label_new" file in https://pan.baidu.com/s/18ZcQQgsr1DFbyUK4YVQ8WQ with the fetech code idrl

**Comment by pianoforte23** [comment]
Hello, thanks for providing the dataset.

I found the issue when running the "train_camouflage_yolov3.py" as shown in the attached images.
I think the data processing code related to masking has some problems.

![test_total_0](https://user-images.githubusercontent.com/62001492/164163179-2f01f053-d612-461f-ab80-7fca68ba6635.png)
![test_total_1](https://user-images.githubusercontent.com/62001492/164163212-262ef7e7-9128-4e66-9096-77bcf6d6eb3a.png)
![test_total_2](https://user-images.githubusercontent.com/62001492/164163227-8cd4c46c-2855-43ad-bd82-f227b283d346.png)
![test_total_3](https://user-images.githubusercontent.com/62001492/164163250-5a060a91-4559-4bd5-8f81-3e81b497153a.png)


Could you please check which part of the code is the problem and reflect the fix?

Thank you.

**Comment by winterwindwang** [comment]
You need to check the adversarial image in "train_camouflage_yolov3.py"( in line 201) first.
`check code: Image.fromarray(np.uint8(imgs[0].cpu().data.numpy().transpose(1, 2, 0))).show()`
 If the resulting image is the same as 
![1650438647](https://user-images.githubusercontent.com/26022270/164171221-ac772141-bf72-4400-bb38-fbb77a449402.jpg)
, that means there is no problem. If not, you should check whether the rendered image has successfully transformed into the training image.

**Comment by pianoforte23** [comment]
My attached images are from line 218.

![image](https://user-images.githubusercontent.com/62001492/164174867-4733ffc4-abf3-40a4-8862-500c6af537fb.png)

I think it saves rendered images as the same result with the line 201.

Please check it again.

Thank you.

**Comment by winterwindwang** [comment]
I have been fixed some problems and pass the test on my computer, please reclone the git repo and test again. If there are still exist problems, please let me know.

**Comment by pianoforte23** [comment]
I recloned your repo and ran "train_camouflage_yolov3.py" again, but the result is the same.

When I run the "generated_and_test.py" code, the rendered images are correct.

So, I think the problem would be in the training code.

The changes I made to run the code and the training history are as follows:

Change: carla.yaml -> path: data/phy_attatck

Training history:
![image](https://user-images.githubusercontent.com/62001492/164204623-b029c069-0cd3-4a69-ba52-d0e44e8a83cb.png)

I think the training process is not working properly because the "grad mean" always comes out as 0.

Please check my problems.

Thank you.

**Comment by winterwindwang** [comment]
Waitting the training process completed and check the trained adversarial texture valid or not.



---Original---
From: ***@***.***&gt;
Date: Wed, Apr 20, 2022 18:08 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [idrl-lab/Full-coverage-camouflage-adversarial-attack] How to use "phy_attack" dataset (Issue #1)




 
I recloned your repo and ran "train_camouflage_yolov3.py" again, but the result is the same.
 
When I run the "generated_and_test.py" code, the rendered images are correct.
 
So, I think the problem would be in the training code.
 
The changes I made to run the code and the training history are as follows:
 
Change: carla.yaml -&gt; path: data/phy_attatck
 
Training history:
 
 
I think the training process is not working properly because the "grad mean" always comes out as 0.
 
Please check my problems.
 
Thank you.

**Comment by zytxxn** [comment]
![image](https://user-images.githubusercontent.com/96740889/164984065-065689ad-2d58-48d1-8344-0d2ba52df9be.png)

**Comment by zytxxn** [comment]
I ran "train_camouflage_yolov3.py" ,but the training process stopped.
![1`AR H()6) $Q%_`TJWXAE](https://user-images.githubusercontent.com/96740889/165017411-1f155db6-d3c8-48a1-a15e-bd8c082c19dd.png)
![%5O {6$@)0314{U~SBE1{6I](https://user-images.githubusercontent.com/96740889/165017410-4033ef3f-a1d2-4de9-9a0f-02195e06efd9.png)
Does anyone have this problem? Could you please tell me how can I solve it?
Thank you very much

**Comment by pianoforte23** [comment]
I found that the dataset I used might be a problem.

I used the "phy_attack" dataset originally used in DAS, and I checked the DAS train process worked well.

However, I found that the result of rendered images from your "generated_and_test.py" is not correct when I use the "phy_attack" dataset directly.

I checked that the result is correct when I use your default sample test dataset (in "carla_dataset/test").

So, could you tell me how to modify the "phy_attack" dataset so that your code runs normally? Or, if possible, could you provide the training dataset you used?

I'd really appreciate it if you reply to my question.
Thank you.

**Comment by winterwindwang** [comment]
> 

DAS dataset also has two part: test data (.npz file) and mask data (.png) file, which can directly used in the my code. So you might used the 3D model (audi_et_te.obj) or faces file (".txt") that proposed in DAS?

**Comment by pianoforte23** [comment]
> > 
> 
> DAS dataset also has two part: test data (.npz file) and mask data (.png) file, which can directly used in the my code. So you might used the 3D model (audi_et_te.obj) or faces file (".txt") that proposed in DAS?

No, I used the 3D model and faces file in your repository.

But I found the reason it didn't work is that the dataset preprocessing was done to run the DAS code.
I checked that your code works properly using the original dataset before DAS preprocessing.

But, I think "utils/datasets_fca.py" has some problems.
May I ask why you set this part as follow image? (line 251~254)
![image](https://user-images.githubusercontent.com/62001492/165227538-037b0e45-6779-478e-a17a-7bf7552e1dea.png)

I found these renderer settings cause some rendering problems as follows:
![image](https://user-images.githubusercontent.com/62001492/165227738-56ad4fc0-7af6-46e0-9a28-cb652d265492.png)

I checked the rendered result is correct when I remove line 251~254 in the "utils/datasets_fca.py" code.
Please check if this code is correct and if I can ignore this part.

Thank you.

**Comment by winterwindwang** [comment]
If the result is fine, you can remove those code.
In the src/readme.md, I recommended to use my modified version of neural renderer, and my code is based on the modified neural renderer. Therefore, if your renderer result is correct, you can move those code.



---Original---
From: ***@***.***&gt;
Date: Tue, Apr 26, 2022 13:28 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [idrl-lab/Full-coverage-camouflage-adversarial-attack] How to use "phy_attack" dataset (Issue #1)




    
DAS dataset also has two part: test data (.npz file) and mask data (.png) file, which can directly used in the my code. So you might used the 3D model (audi_et_te.obj) or faces file (".txt") that proposed in DAS?
  
No, I used the 3D model and faces file in your repository.
 
But I found the reason it didn't work is that the dataset preprocessing was done to run the DAS code.
 I checked that your code works properly using the original dataset before DAS preprocessing.
 
But, I think "utils/datasets_fca.py" has some problems.
 May I ask why you set this part as follow image? (line 251~254)
 
 
I found these renderer settings cause some rendering problems as follows:
 
 
I checked the rendered result is correct when I remove line 251~254 in the "utils/datasets_fca.py" code.
 Please check if this code is correct and if I can ignore this part.
 
Thank you.

**Comment by pianoforte23** [comment]
> If the result is fine, you can remove those code. In the src/readme.md, I recommended to use my modified version of neural renderer, and my code is based on the modified neural renderer. Therefore, if your renderer result is correct, you can move those code.
> […](#)
> ---Original--- From: ***@***.***&gt; Date: Tue, Apr 26, 2022 13:28 PM To: ***@***.***&gt;; Cc: ***@***.******@***.***&gt;; Subject: Re: [idrl-lab/Full-coverage-camouflage-adversarial-attack] How to use "phy_attack" dataset (Issue #1) DAS dataset also has two part: test data (.npz file) and mask data (.png) file, which can directly used in the my code. So you might used the 3D model (audi_et_te.obj) or faces file (".txt") that proposed in DAS? No, I used the 3D model and faces file in your repository. But I found the reason it didn't work is that the dataset preprocessing was done to run the DAS code. I checked that your code works properly using the original dataset before DAS preprocessing. But, I think "utils/datasets_fca.py" has some problems. May I ask why you set this part as follow image? (line 251~254) I found these renderer settings cause some rendering problems as follows: I checked the rendered result is correct when I remove line 251~254 in the "utils/datasets_fca.py" code. Please check if this code is correct and if I can ignore this part. Thank you. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

Thank you very much :)

**Comment by onetrance** [comment]
> 

Excuse me.The "train_new" needs the ".png" but the dataset in "phy_attack" in DAS is the ".npz".Now I have downloaded the "phy_attack" with ".npz".How can I convert it?

**Comment by qwc0121** [comment]
> > > 
> > 
> > 
> > No, I didn't solve it yet :(
> 
> Hey, we now provide the "train_label_new" file in https://pan.baidu.com/s/18ZcQQgsr1DFbyUK4YVQ8WQ with the fetech code `idrl`

Hello, may I ask if the texture you finally reproduced can achieve the effect similar to the texture provided by the original author in the yolov5x test? And I found that the data set provided did not have annotations for the test set
